include/llbuild/Core/BuildEngine.h

```diff
@@ -34,8 +34,39 @@
 namespace llbuild {
 namespace core {
 
-// FIXME: Need to abstract KeyType;
-typedef std::string KeyType;
+struct KeyType {
+  std::string function;
+  std::vector<uint8_t> args;
```

Comment:
Should we do a custom allocator for this so we can be more efficient? We probably want these to be immutable, and could do a single heap allocation and end up with substantially less memory use than this (std::string and std::vector are probably each 3 words or more?)